tensorflow/core/user_ops/chol_grad.cc

```diff
@@ -0,0 +1,115 @@
+//Copyright (c) 2016, Alexander G. de G. Matthews and James Hensman. All rights reserved.
+
+//Based on code for blocked Cholesky reference mode differentiation by Iain Murray.
+//For an explanation see "Differentiation of the Cholesky algorithm" by Iain Murray http://arxiv.org/abs/1602.07527.
+
+#include "tensorflow/core/framework/op.h"
+#include "third_party/eigen3/Eigen/Core"
+
+REGISTER_OP("CholeskyGrad").Input("l: T").Input("l_bar: T").Output("a_bar: T").Attr( "T: {float, double}").Doc("Cholesky backpropagation where l is output of Cholesky algorithm and f is gradient of some loss wrt l");
+
+#include "tensorflow/core/framework/op_kernel.h"
+
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/framework/types.h"
```

Comment:
Please put the op registration and docstring in tensorflow/core/ops/linalg_ops.cc.
Please put one argument per line (newline before each '.').

I would recommend not using the notation from Murray's paper to name the input and output arguments. Perhaps name them something like "cholesky", "cholesky_grad", and "matrix_grad". 

Please make your docstring a bit more detailed (see the registration of cholesky in ops/linalg_ops.cc for example). Also, get rid of references to undefined entities "f" and "l"; I think it suffices to say that the function calculates the gradient w.r.t. to the input matrix, given the gradient w.r.t. its Cholesky factorization.